nixos/modules/services/web-servers/quakejs.nix

```diff
@@ -0,0 +1,207 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkEnableOption mkOption literalExpression;
+  inherit (lib.types)
+    lines
+    submodule
+    bool
+    uniq
+    str
+    port
+    ;
+  inherit (lib.lists) optionals;
+
+  cfg = config.services.quakejs;
+  format = pkgs.formats.json { };
+
+in
+{
+  options.services.quakejs = {
+
+    enable = mkEnableOption "Game server for QuakeJS";
+
+    hostname = mkOption {
+      default = "localhost";
+      type = uniq str;
+      example = ''example.com'';
+      description = ''
+        Hostname to use. It should be FQDN.
+      '';
+    };
+
+    eula = mkOption {
+      type = bool;
+      description = ''
+        By changing this option to true you confirm that you own a copy of Quake 3 Arena,
+        and that you agree to minecrafts EULA.
+      '';
+      default = false;
+    };
+
+    openFirewall = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Open firewall for web client, dedicated and content server.
+      '';
+    };
+
+    settings = mkOption {
+      type = submodule { freeformType = format.type; };
+      description = ''
+        Configuration for the web game client.
+      '';
+      default = {
+        content = "content.quakejs.com";
+        port = 8080;
+      };
+      example = literalExpression ''
+        {
+          content = "content.quakejs.com";
+          port = 8081;
+        }
+      '';
+    };
+
+    content-server = {
+
+      enable = mkEnableOption "Content server for QuakeJS";
+
+      hostname = mkOption {
+        default = "localhost";
+        type = uniq str;
+        example = ''example.com'';
+        description = ''
+          Hostname to use. It should be FQDN.
+        '';
+      };
+
+      settings = mkOption {
+        type = submodule { freeformType = format.type; };
+        description = ''
+          Configuration for the content server.
+        '';
+        default = {
+          root = "/var/lib/quakejs";
+          port = 8081;
+        };
+      };
+
+    };
+
+    dedicated-server = {
+
+      enable = mkEnableOption "Dedicated server for QuakeJS";
+
+      port = mkOption {
+        type = port;
+        default = 27960;
+        description = ''
+          Port for the server to listen on.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = lines;
+        description = ''
+          Extra configuration options. Note that options changed via RCON will not be persisted. To list all possible options, use “cvarlist 1” via RCON.
+        '';
+        default = ''
```

Comment:
Shouldn't we set this in config? Otherwise people setting one line, need to repeat all of that